tx_pipeline/Cargo.toml

```diff
@@ -0,0 +1,20 @@
+[package]
+name = "miden-tx-pipeline"
```

Comment:
I wonder if we should name this `miden-block-producer` (and the containing directory `block_prodcuder`). I think this captures better the purpose of this component (collect transactions and output blocks). If we decide later on to break out txpool, we could do that and still retain the name `miden-block-producer`.

Ideally, we'd come up with a single-word term for this - but I don't love `miden-miner` (or `miden-producer`) and nothing else comes to mind. Maybe `miden-operator` but to me `miden-block-producer` is a bit more clear.